src/rm.rs

```
@@ -3,12 +3,12 @@ use policy_evaluator::policy_fetcher::store::{PolicyPath, Store};
 use std::path::PathBuf;
 
 pub(crate) fn rm(uri_or_sha_prefix: &str) -> Result<()> {
-    let uri = crate::utils::map_path_to_uri(uri_or_sha_prefix)?;
+    let uri = crate::utils::get_uri(&uri_or_sha_prefix.to_string())?;
 
     let store = Store::default();
     let policy_path = store.policy_full_path(&uri, PolicyPath::PrefixAndFilename)?;
     std::fs::remove_file(&policy_path)
-        .map_err(|err| anyhow!("could not delete policy {}: {}", uri, err))?;
+        .map_err(|_| anyhow!("Failed to remove policy {}: Cannot find policy ", uri))?;
```

If we do that, we will hide other possible errors in the std lib function used to remove the file. 

I was reviewing the code more carefully and notice something. Let's take a look in the `get_uri` function:

https://github.com/kubewarden/kwctl/blob/d49f285e421e2f8ef9d709d894f856bfb2eff413/src/utils.rs#L27-L54

Note that the `get_uri` function is hitting the `or_else` which will generate a valid policy URI even if the policy does not exists. That's why the code here is reaching the line that tries to remove the file. What you have to do is something similar of what `map_path_to_uri` does. Change the code to check if the uri returned by the `get_uri` actually exists in the store _before_ calling the `remove_file` function.  You may want to use the `get_policy_by_uri` method for that:

https://github.com/kubewarden/policy-fetcher/blob/34cdd3b9136fcbd97fbaf51ec94117ff6de9a976/src/store/mod.rs#L170

Sorry for this ping-pong and thanks for your patience! :heart: